fastlane/Fastfile

```diff
@@ -116,17 +116,24 @@ end
 
 lane :merge_release_to_main do
   ensure_git_status_clean
-  sh('git checkout main')
-  sh('git pull')
 
-  # Grep all remote release branches and ensure there's only one
-  release_branches = sh(command: 'git branch -a', log: false).delete(' ').split("\n").grep(%r(origin/.*release/))
-  UI.user_error!("Expected 1 release branch, found #{release_branches.size}") if release_branches.size != 1
+  release_branch =
+    if is_ci
+      current_branch
+    else
+      release_branches = sh(command: 'git branch -a', log: false).delete(' ').split("\n").grep(%r(origin/.*release/))
+      UI.user_error!("Expected 1 release branch, found #{release_branches.size}") if release_branches.size != 1
+      release_branches.first
+    end
+
+  UI.user_error!("`#{release_branch}`` branch does not match the release branch pattern: `release/*`") unless release_branch.start_with?('release/')
+
+  sh('git checkout origin/main')
+  sh('git pull origin main')
 
   # Merge release branch to main. For more info, read: https://notion.so/iOS-Branching-Strategy-37c10127dc26493e937769d44b1d6d9a
-  sh("git merge #{release_branches.first} --ff-only")
-  UI.user_error!('Not pushing changes') unless prompt(text: 'Will push changes. All looking good?', boolean: true)
-  sh('git push')
+  sh("git merge #{release_branch} --ff-only")
+  sh('git push origin HEAD:main')
```

Comment:
I assume it will work similar to `merge main to develop`, so also should be tested on release.